nodecontrol/NodeController.go

```diff
@@ -111,26 +111,26 @@ func (nc NodeController) stopProcesses() (kmdAlreadyStopped bool, err error) {
 	return
 }
 
-func killPID(pid int) error {
+func killPID(pid int) (killed bool, err error) {
```

Comment:
Instead of returning killed, and deleting the pid file, why not pass the pid file path to killPID, and delete the file?

Reply:
I think that the `killPID` should be focused on *just* killing the process ( without worrying how the pid was obtained ).
The caller, who have the knowledge of how the pid was obtained, can delete the file.
( either directly or indirectly ).